services/scheduler/src/main/java/org/mule/service/scheduler/internal/config/ThreadPoolsConfig.java

```diff
@@ -0,0 +1,217 @@
+ * LICENSE.txt file.
+ */
+package org.mule.service.scheduler.internal.config;
+
+import static java.io.File.separator;
+import static java.lang.Integer.parseInt;
+import static java.lang.Runtime.getRuntime;
+import static java.lang.System.getProperty;
+import static java.util.regex.Pattern.compile;
+import static org.mule.runtime.core.api.config.MuleProperties.MULE_HOME_DIRECTORY_PROPERTY;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.core.api.DefaultMuleException;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import org.slf4j.Logger;
+
+/**
+ * Bean that contains the thread pools configuration for the runtime.
+ *
+ * @since 4.0
+ */
+public class ThreadPoolsConfig {
+
+  private static final Logger logger = getLogger(ThreadPoolsConfig.class);
+
+  public static final String PROP_PREFIX = "SchedulerService.";
+
+  private static final String NUMBER_OR_C_REGEXP = "([0-9]+(\\.[0-9]+)?)|(C)";
+  private static final Pattern POOLSIZE_PATTERN =
+      compile("(" + NUMBER_OR_C_REGEXP + ")?(\\s*[-+\\/*]\\s*(" + NUMBER_OR_C_REGEXP + ")?)*");
+
+  /**
+   * Loads the configuration from the {@code &#123;mule.home&#125;/conf/scheduler-pools.conf} file.
+   * 
+   * @return The loaded configuration, or the default if the file is unavailable.
+   * @throws MuleException for any trouble that happens while parsing the file.
+   */
+  public static ThreadPoolsConfig loadThreadPoolsConfig() throws MuleException {
+    File muleHome =
+        getProperty(MULE_HOME_DIRECTORY_PROPERTY) != null ? new File(getProperty(MULE_HOME_DIRECTORY_PROPERTY)) : null;
+
+    final ThreadPoolsConfig config = new ThreadPoolsConfig();
+
+    if (muleHome == null) {
+      logger.info("No " + MULE_HOME_DIRECTORY_PROPERTY + " defined. Using default values for thread pools.");
+      return config;
+    }
+
+    File defaultConfigFile = new File(muleHome, "conf" + separator + "scheduler-pools.conf");
+    if (!defaultConfigFile.exists()) {
+      logger.info("No thread pools config file found. Using default values.");
+      return config;
+    }
+
+    logger.info("Loading thread pools configuration from " + defaultConfigFile.getPath());
+
+    final Properties properties = new Properties();
+    try (final FileInputStream configIs = new FileInputStream(defaultConfigFile)) {
+      properties.load(configIs);
+    } catch (IOException e) {
+      throw new DefaultMuleException(e);
+    }
+
+    ScriptEngineManager manager = new ScriptEngineManager();
+    ScriptEngine engine = manager.getEngineByName("js");
+    engine.put("C", config.cores);
+
+    config.setGracefulShutdownTimeoutSeconds(resolveNumber(properties, PROP_PREFIX + "gracefulShutdownTimeoutSeconds"));
```

Comment:
Compose these strings with constants for "cpuLight" "IO" "coreThreadPoolSize" etc.